pragma-node/src/infra/repositories/onchain_repository.rs

```diff
@@ -286,9 +405,25 @@ pub async fn get_sources_and_aggregate(
         return Err(InfraError::NotFound);
     }
 
-    let aggregated_price = raw_entries.first().unwrap().aggregated_price.clone();
-    let entries: Vec<OnchainEntry> = raw_entries.into_iter().map(From::from).collect();
-    Ok((aggregated_price, entries))
+    let mut result: Vec<AggPriceAndEntries> = Vec::new();
```

Comment:
It deserves also its own function that builds the result + comment

Reply:
done